src/stan/services/sample/fixed_param.hpp

```diff
@@ -65,15 +66,13 @@ int fixed_param(Model& model, const stan::io::var_context& init,
   writer.write_sample_names(s, sampler, model);
   writer.write_diagnostic_names(s, sampler, model);
 
-  clock_t start = clock();
-
+  auto start = std::chrono::steady_clock::now();
   util::generate_transitions(sampler, num_samples, 0, num_samples, num_thin,
                              refresh, true, false, writer, s, model, rng,
                              interrupt, logger);
-  clock_t end = clock();
-
-  double sampleDeltaT = static_cast<double>(end - start) / CLOCKS_PER_SEC;
-  writer.write_timing(0.0, sampleDeltaT);
+  auto end = std::chrono::steady_clock::now();
+  std::chrono::duration<double> sample_delta_t = end - start;
```

Comment:
The way you coded it is along the lines of the example on en.cppreference.com. I still do not see in the definition of the standard that one "tick" is equal to one second. Could we used instead this (if it works):

`std::chrono::seconds sample_delta_t = end - start;`

This way we make sure that its indeed seconds which we report.

See https://en.cppreference.com/w/cpp/chrono/duration and there "helper types".

Although that could report time as full seconds only... maybe test that.

Another option is to make use of the "period" member type which is part of the duration class. The period member type defines the number of seconds per tick. Maybe this is over-cautious, but I don't see in the standard that one tick is equal to one second on all systems.

Reply:
It says

It consists of a count of ticks of type Rep and a tick period, where the tick period is a compile-time rational constant representing the number of seconds from one tick to the next.

I will check if there is a way to explicitly mention seconds

Reply:
I know... but the duration you use is just the number of ticks elapsed. You do not know that their base unit are seconds. So what is 1 tick is not defined by the standard, but system specific in my understanding of the standard.

If we use `std::chrono::seconds` then it will be guaranteed that we count in seconds.

Reply:
`std::chrono::seconds` is actually just `std::chrono::duration<int64_t, std::ratio<1,1>>`.

So the only other way of doing this would be:
```cpp
auto a = std::chrono::steady_clock::now();
// ... work ...
auto b = std::chrono::steady_clock::now();
double diff = std::chrono::duration_cast<std::chrono::milliseconds>(b-a).count()/1000.0;
```
But that feels cumbersome. But if you prefer that I will change it.

If we do `std::chrono::duration_cast<std::chrono::seconds>` that returns just the integer part of the number of seconds.

This is the implementation of duration.
```
 /// duration
    template<typename _Rep, typename _Period>
      struct duration
      {
	typedef _Rep						rep;
	typedef _Period 					period;
```
and `_Rep` is `double` in our case. And period is 1/(ticks per sec).

Reply:
I feared that we would loose below second accuracy if we go with the seconds thing.

Let me have one more read of the C++ ref and then we will likely proceed with what you have implemented. It appears to me that the C++ ref docs are bad in this case unfortunately.

Reply:
I went through the C++ docs once more and googled around. The issue is still that a "tick" is never defined to be equal "one second". This is left as an implementation detail. One tick is probably commonly defined as one second, but the standard does not define it.

I think you suggestion with milliseconds is best if want to maintain sub-second accuracy. You can probably write it in an easier way, since the duration_cast should happen implicitly. So maybe try the less cumbersome

`double diff = std::chrono::milliseconds(b-a).count()/1000.0;`

I would expect that to work as well, but just give it a try.

This is the best we can do from what I see to get well defined time units. It's a shortcoming of the C++ standard here, I am afraid.

Thanks for bearing with me here.

Reply:
I still think that text here does say that: https://en.cppreference.com/w/cpp/chrono/duration

Quote:
It consists of a count of ticks of type Rep and a tick period, **where the tick period is a compile-time rational constant representing the number of seconds from one tick to the next**.

In the old time.h way, that means that period is basically something close to `1/CLOCKS_PER_SEC`. So the fraction of a second that represent one tick. Its probably written in a bizarre way.

I will try to see if I can do something close to. 
`double diff = std::chrono::milliseconds(b-a).count()/1000.0;`

I do have to note that this only helps us in "code readability", as std::chrono::milliseconds only applies the std::millli [ratio](https://en.cppreference.com/w/cpp/numeric/ratio/ratio) to the duration at compile time. So you are still counting on duration being returned in seconds.

> Thanks for bearing with me here.

I have no problem being cautious.

Reply:
> It consists of a count of ticks of type Rep and a tick period, where the tick period is a compile-time rational constant representing the number of seconds from one tick to the next.

Correct. So what you could do as an alternative is to take from an instance of `std::chrono::duration<double>` the `_Preiod` definition and multiply the number which we get with `.count()` by the ratio defined by `_Period`. It's an implementation detail that `_Period` is unity (meaning it is equal to one second), but that equality ( 1 tick = 1 second ) is not spelled out anywhere.

The solution with milliseconds is much cleaner from my perspective and it results in well defined behaviour... much better!